flake.nix

```diff
@@ -0,0 +1,91 @@
+{
+  description = "A Nix-flake for poetry based python development ";
+
+  inputs = {
+    flake-utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    poetry2nix = {
+      url = "github:nix-community/poetry2nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+      inputs.flake-utils.follows = "flake-utils";
+    };
+  };
+
+  # has standard of which keys
+  outputs = {self, ...} @ inputs:
+    inputs.flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = import inputs.nixpkgs {localSystem = {inherit system;};};
+      inherit (pkgs) lib;
+
+      # Project metadata
+      pyproject = builtins.fromTOML (builtins.readFile ./pyproject.toml);
+
+      # Actual python project
+      inherit
+        (inputs.poetry2nix.lib.mkPoetry2Nix {inherit pkgs;})
+        mkPoetryApplication
+        mkPoetryEnv
+        ;
+
+      # Hardcode python version:
+      python = pkgs.python312;
+      poetryConfig = {
+        inherit python;
+        projectDir = self;
+        preferWheels = true;
+        groups = ["rest" "bigquery_search"];
+      };
+      pythonApp = pkgs.writeShellApplication {
+        inherit (pyproject.tool.poetry) name;
+        runtimeInputs = [
+          # The actual Poetry app:
+          (mkPoetryApplication poetryConfig)
+          .dependencyEnv
+        ];
+        text = "uvicorn nks_ds.rest.main:app --host 0.0.0.0 --port 8080";
```

Comment:
This will fail at run-time (not build time) unless your app does/supports the same thing...